src/RIGID/fix_rigid.cpp

```diff
@@ -2523,9 +2523,17 @@ int FixRigid::pack_exchange(int i, double *buf)
   buf[2] = displace[i][0];
   buf[3] = displace[i][1];
   buf[4] = displace[i][2];
-  if (!extended) return 5;
+
```

Comment:
Can you attach an input deck example to reproduce this?
We also need to check if the same issue applies to other fix rigid variants.

Reply:
The example I had was quite large, but now that I narrowed down the origin of the bug I can try and put together a minimal script to reproduce the issue (weird stress values along MPI grid lines after exchanging atoms).

The issue does not affect fix rigid/small (it doesn't return if _extended_) and the other rigid methods should inherit the exchange methods. But I'll double check.

Reply:
Using the current develop branch, this script should reproduce the bug when run on 4 processors [in.melt.txt](https://github.com/lammps/lammps/files/11023140/in.melt.txt). It creates a small circular rigid body in one quadrant and slowly moves it through a LJ melt. The dump file includes a column for the XX component of the virial from fix rigid. On timestep 900, there should be two non-zero values inside the fluid (particle 329 and 222, rendered below). With this PR, only rigid particles have stress contributions from fix rigid.


![image](https://user-images.githubusercontent.com/63308469/226469251-0a8ef8de-5961-431d-b7e7-57cf72854a6d.png)